packages/core/src/resource.ts

```diff
@@ -180,27 +212,46 @@ export abstract class Resource extends Route implements ResourceInterface {
         );
 
     public del = <
-        R = any,
-        D extends ObjectMap = any,
-        Params extends Kwargs | null = Kwargs
+        TResponse = EmptyPayload,
+        TPayload extends TPostPayload = TPostPayload,
+        TParams extends Params = Params
     >(
-        kwargs: Params | null = null,
-        data: D | string | null = null,
+        kwargs: TParams | null = null,
+        data: TPayload | string | null = null,
         query: Query | null = null,
         attachments: Attachments | null = null,
         requestConfig: RequestConfig | null = null
-    ): Promise<R> | any =>
-        this._post<R, D, Params>(
+    ): Promise<TResponse> =>
+        this.delete<TResponse, TPayload, TParams>(
+            kwargs,
+            data,
+            query,
+            attachments,
+            requestConfig
+        );
+
+    public delete = <
```

Comment:
`delete` is a reserved name so don't add this as it may cause some issues downstream

just fix the method name on `del`

Reply:
this actually happened, we may be passed this era but in one of our projects the bundler did not like that name at all

Reply:
ok, will change